server/src/main/java/org/spine3/server/storage/StorageFactorySwitch.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.storage;
+
+import com.google.common.base.Supplier;
+import org.spine3.server.storage.memory.InMemoryStorageFactory;
+import org.spine3.util.Environment;
+
+import javax.annotation.Nullable;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * The {@code Supplier} of {@code StorageFactory} that takes into account
+ * if the code runs under tests.
+ *
+ * <h2>Test mode</h2>
+ * <p>Under tests  this class returns {@link InMemoryStorageFactory} if
+ * a {@code Supplier} for tests was not set via {@link #init(Supplier, Supplier)}.
+ *
+ * <h2>Production mode</h2>
+ * <p>In production mode this class obtains the instance obtained
```

Comment:
`...obtains the instance obtained...` looks weird.

I think `obtains the instance provided by the` will do the job.